### ISSUE TITLE
org.cactoos.map.IterableAsMap

### ISSUE THREAD

**Issue by yegor256** [yegor256/cactoos#116]
Let's create a new package and class.

**Comment by Timmeey** [comment]
@yegor256 like this?
```java
public final class IterableAsMap<X,Y,Z>{
  private final Iterable<X> iterable;
  private final Func<X,Set<Y,Z>> mapping;
  public IterableAsMap(Iterable<X> iterable, Func<X,Set<Y,Z>> func){}
```

**Comment by yegor256** [comment]
@Timmeey like this:

```java
        MatcherAssert.assertThat(
            "Can't convert iterable to map",
            new IterableAsMap<>(
                new AbstractMap.SimpleEntry<>(0, "hello, "),
                new AbstractMap.SimpleEntry<>(1, "world!")
            ),
            Matchers.hasEntry(
                Matchers.equalTo(0),
                Matchers.startsWith("hello")
            )
        );

```

I'm working on it now...

**Comment by yegor256** [comment]
fixed in #118

**Comment by yegor256** [comment]
@rultor release, tag is `0.4`
